phpunit/class-rest-sidebars-controller-test.php

```diff
@@ -0,0 +1,410 @@
+		global $wp_registered_sidebars;
+		update_option(
+			'sidebars_widgets',
+			array(
+				$id => $widgets,
+			)
+		);
+		$wp_registered_sidebars[ $id ] = array_merge(
+			array(
+				'id'            => $id,
+				'before_widget' => '',
+				'after_widget'  => '',
+				'before_title'  => '',
+				'after_title'   => '',
+			),
+			$attrs
+		);
+
+		global $wp_registered_widgets;
+		foreach ( $wp_registered_widgets as $wp_registered_widget ) {
+			if ( is_array( $wp_registered_widget['callback'] ) ) {
+				$wp_registered_widget['callback'][0]->_register();
+			}
+		}
+	}
+
+	/**
+	 *
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey( '/__experimental/sidebars', $routes );
+		$this->assertArrayHasKey( '/__experimental/sidebars/(?P<id>[\w-]+)', $routes );
+	}
+
+	/**
+	 *
+	 */
+	public function test_context_param() {
+	}
+
+	/**
+	 *
+	 */
+	public function test_get_items() {
+		$request  = new WP_REST_Request( 'GET', '/__experimental/sidebars' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( array(), $data );
+	}
+
+	/**
+	 *
+	 */
+	public function test_get_items_basic_sidebar() {
+		$this->setup_sidebar(
+			'sidebar-1',
+			array(
+				'name' => 'Test sidebar',
+			)
+		);
+
+		$request  = new WP_REST_Request( 'GET', '/__experimental/sidebars' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals(
+			array(
+				array(
+					'id'          => 'sidebar-1',
+					'name'        => 'Test sidebar',
+					'description' => '',
+					'status'      => 'active',
+					'widgets'     => array(),
+				),
+			),
+			$data
+		);
+	}
+
+	/**
+	 *
+	 */
+	public function test_get_items_active_sidebar_with_widgets() {
+		$this->setup_widget(
+			'widget_rss',
+			1,
+			array(
+				'title' => 'RSS test',
+			)
+		);
+		$this->setup_widget(
+			'widget_text',
+			1,
+			array(
+				'text' => 'Custom text test',
+			)
+		);
+		$this->setup_sidebar(
+			'sidebar-1',
+			array(
+				'name' => 'Test sidebar',
+			),
+			array( 'text-1', 'rss-1' )
+		);
+
+		$request  = new WP_REST_Request( 'GET', '/__experimental/sidebars' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals(
+			array(
+				array(
+					'id'          => 'sidebar-1',
+					'name'        => 'Test sidebar',
+					'description' => '',
+					'status'      => 'active',
+					'widgets'     => array(
+						array(
+							'id'           => 'text-1',
+							'settings'     => array(
+								'text' => 'Custom text test',
+							),
+							'id_base'      => 'text',
+							'widget_class' => 'WP_Widget_Text',
+							'name'         => 'Text',
+							'description'  => 'Arbitrary text.',
+							'number'       => 1,
+							'rendered'     => '			<div class="textwidget">Custom text test</div>' . "\n		",
```

Comment:
What causes all the empty space? I wonder if it makes sense to `trim` the rendered form?